extensions/vscode-api-tests/src/singlefolder-tests/untitled.languagedetection.test.ts

```diff
@@ -0,0 +1,74 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as vscode from 'vscode';
+import { assertNoRpc, closeAllEditors } from '../utils';
+
+function sleep(ms: number) {
+	return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+suite('vscode - untitled automatic language detection', () => {
```

Comment:
Isn't this test depending on language detection being enabled? If so should it be enabled as a workspace setting here: `extensions/vscode-api-tests/testWorkspace/.vscode/settings.json`

Reply:
Yeah in the test you can see its applying the LanguageDetection setting. However, your idea is better since having this setting on in other scenarios want do any harm.